presto-main/src/main/java/com/facebook/presto/sql/planner/BasePlanFragmenter.java

```diff
@@ -223,6 +201,31 @@ public PlanNode visitTableFinish(TableFinishNode node, RewriteContext<FragmentPr
         return context.defaultRewrite(node, context.get());
     }
 
+    @Override
+    public PlanNode visitSequence(SequenceNode node, RewriteContext<FragmentProperties> context)
+    {
+        // Since this is topologically sorted by the LogicalCtePlanner, need to make sure that execution order follows
+        // Can be optimized further to avoid non dependents from getting blocked
+        int n = node.getCteProducers().size();
+        checkArgument(n >= 1, "Sequence Node has 0 cte producers");
```

Comment:
nit: since "CTE" is an abbreviation, can we use it consistently capitalized in error and warning messages displayed to the user